relayer/src/event/rpc.rs

```diff
@@ -141,31 +138,31 @@ pub fn get_all_events(
 
             for abci_event in &tx_result.result.events {
                 if query == queries::ibc_client().to_string() {
-                    if let Some(mut client_event) = ClientEvents::try_from_tx(abci_event) {
+                    if let Some(mut client_event) = client_events::try_from_tx(abci_event) {
                         client_event.set_height(height);
                         tracing::trace!("extracted ibc_client event {}", client_event);
                         vals.push((height, client_event));
                     }
                 }
                 if query == queries::ibc_connection().to_string() {
-                    if let Some(mut conn_event) = ConnectionEvents::try_from_tx(abci_event) {
+                    if let Some(mut conn_event) = connection_events::try_from_tx(abci_event) {
                         conn_event.set_height(height);
                         tracing::trace!("extracted ibc_connection event {}", conn_event);
                         vals.push((height, conn_event));
                     }
                 }
                 if query == queries::ibc_channel().to_string() {
-                    if let Some(mut chan_event) = ChannelEvents::try_from_tx(abci_event) {
+                    if let Some(mut chan_event) = channel_events::try_from_tx(abci_event) {
                         chan_event.set_height(height);
                         let _span = tracing::trace_span!("ibc_channel event").entered();
                         tracing::trace!("extracted {}", chan_event);
                         if matches!(chan_event, IbcEvent::SendPacket(_)) {
+                            // We are interested in the tag "hash" of an event with type "tx",
+                            // but this is trace level, so print the whole list of events.
                             // Should be the same as the hash of tx_result.tx?
-                            if let Some(hash) =
-                                events.get("tx.hash").and_then(|values| values.get(0))
-                            {
-                                tracing::trace!(event = "SendPacket", "tx hash: {}", hash);
-                            }
+                            // TODO: use a helper function to filter to just the interesting
```

Comment:
Let's keep the tx hash trace if possible.

ALso update the comments for this method.